beacon-chain/blockchain/forkchoice/process_attestation.go

```diff
@@ -96,31 +97,32 @@ func (s *Store) OnAttestation(ctx context.Context, a *ethpb.Attestation) (uint64
 		return 0, err
 	}
 
-	// Process aggregated attestation in the queue every `slot/2` duration,
-	// this allows the incoming attestation to aggregate and avoid
-	// excessively processing individual attestations.
-	if halfSlot(baseState.GenesisTime) {
-		s.attsQueueLock.Lock()
-		for root, a := range s.attsQueue {
-			log.WithFields(logrus.Fields{
-				"AggregatedBitfield": fmt.Sprintf("%b", a.AggregationBits),
-				"Root":               fmt.Sprintf("%#x", root),
-			}).Debug("Updating latest votes")
-
-			// Use the target state to to validate attestation and calculate the committees.
-			indexedAtt, err := s.verifyAttestation(ctx, baseState, a)
-			if err != nil {
-				return 0, err
-			}
+	s.attsQueueLock.Lock()
+	for root, a := range s.attsQueue {
+		log.WithFields(logrus.Fields{
+			"AggregatedBitfield": fmt.Sprintf("%b", a.AggregationBits),
+			"Root":               fmt.Sprintf("%#x", root),
+		}).Debug("Updating latest votes")
+
+		// Use the target state to to validate attestation and calculate the committees.
+		indexedAtt, err := s.verifyAttestation(ctx, baseState, a)
+		if err != nil {
+			return 0, err
+		}
 
-			// Update every validator's latest vote.
-			if err := s.updateAttVotes(ctx, indexedAtt, tgt.Root, tgt.Epoch); err != nil {
-				return 0, err
-			}
-			delete(s.attsQueue, root)
+		// Update every validator's latest vote.
+		if err := s.updateAttVotes(ctx, indexedAtt, tgt.Root, tgt.Epoch); err != nil {
+			return 0, err
+		}
+
+		// Mark attestation as seen we don't update votes when it appears in block.
+		if err := s.setSeenAtt(a); err != nil {
+			return 0, err
 		}
-		s.attsQueueLock.Unlock()
+
+		delete(s.attsQueue, root)
 	}
+	s.attsQueueLock.Unlock()
```

Comment:
If this func errors in the for loop, the lock will never be released. Prefer defer here.